module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatterFactory.php

```
@@ -67,6 +66,13 @@ class RecordDataFormatterFactory implements FactoryInterface
      */
     protected $authorOrder = ['primary' => 1, 'corporate' => 2, 'secondary' => 3];
 
+    /**
```

The authorOrder property above seems to be orphaned.

---

I left it for BC, because someone might have used it in their version of the factory. I added a deprecation tag but can remove it if you prefer that.

---

I think we need to make a decision about backward compatibility, since in the current state of things, I think we only have partial compatibility, which is probably more confusing than choosing either a hard break or more robust compatibility.

For example: historically, I have customized the record data formatter by overriding/extending the `getDefault*Specs` methods. As currently written, those methods are no longer used, so those customizations will just be broken and lost without warning. Fully implementing compatibility with those methods is possible but might be a little complicated. At very least, we might want to check to see if those methods are defined and throw a deprecation warning, or even a hard exception.

The RecordDataFormatter has evolved more than just about any other part of the code, and for good reasons -- but there are now a lot of different ways of doing things. I think the best decision may be to decide on some best practices and warn people who are doing things in outdated/less desirable ways. Maybe we need to have a Community Call discussion about this since it could be quite impactful.

In any case, I'm definitely interested in moving forward with the work here, as I think this will make things better -- but we may want to leave this open for a while to make sure we're doing it in the least disruptive way that we reasonably can.

---

Please note that the `getDefault*Specs` methods are still being used here https://github.com/ThoWagen/vufind/blob/bf8b9eb1b9e4a05fcb945c301f73f213622ff957/module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatterFactory.php#L103-L116 . So as long as you don't have overwritten the `__invoke` method and only the `getDefault*Specs` methods, this should still work. 

But I'm certainly open to removing BC here completely.

---

The problem is that a common pattern is to call the parent method from the local custom method, and we've removed the parent method -- so I think if we want full compatibility, we need to restore all of those methods as deprecated wrappers around the spec object, and if we don't, we should probably throw an error here (since otherwise we'll just break with "undefined method" fatal errors on the `parent::` call in some, possibly most, cases).

---

(Given that you've gone to the trouble of incorporating most of the backward compatibility, it's probably worth taking this extra step... but it may be worth throwing some deprecation warnings that will show up in logs to encourage people to revisit any customizations they've made here).

---

Ah yes, of course! We have completely overwritten those methods, so I didn't think of that. I will add the wrappers with deprecation warnings.

---

Done